### DISCUSSION TITLE
[Core Protocol] How is precompile logic disputed in Cannon?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [ethereum-optimism/developers#421] [Core Protocol]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [X] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Precompiles are implemented natively in the client software, e.g. in `go-ethereum` they are implemented [here](https://github.com/ethereum/go-ethereum/blob/b6474e9f90c88003de7d7eb993ddb5f60133172e/core/vm/contracts.go#L4).

How does [Cannon](https://docs.optimism.io/stack/protocol/fault-proofs/cannon) (the default Fault Proof VM in OP Stack) work when there is a dispute for a transaction that calls a precompile, e.g. the `ecrecover` located at address `0x01`?

How can the onchain `MIPS.sol` interpreter re-run precompile logic?

### Additional Information

I've read the docs on Fault Proofs:

https://docs.optimism.io/stack/protocol/fault-proofs/overview

**Comment by opfocus** [comment]
https://specs.optimism.io/experimental/fault-proof/index.html#type-6-global-precompile-key      _This preimage key can be used to avoid running expensive precompile operations in the program._    
This may be relevant content, if it doesn't answer your question please ask your question on github for that documentation

**Reply by PaulRBerg** [reply]
Thank you for sharing that link to the specs. It is indeed relevant.

I have moved this discussion to the specs repository:

https://github.com/ethereum-optimism/specs/discussions/211
